python/letsql/backends/let/tests/test_execute.py

```diff
@@ -374,3 +374,17 @@ def test_expr_over_same_table_multiple_times(con, parquet_dir, other_con):
 
     con.drop_table(ls_table_name)
     con.drop_table(batting_name)
+
+
+def test_register_with_different_name(con, csv_dir):
+    batting_path = csv_dir.joinpath("batting.csv")
+    table_name = "batting"
+
+    duckdb_con = ibis.duckdb.connect()
+    t = duckdb_con.register(batting_path, table_name=table_name)
+    con.register(t, table_name=f"{duckdb_con.name}_{table_name}")
+
+    t = con.table(f"{duckdb_con.name}_{table_name}")
+    assert t.execute() is not None
```

Comment:
I've added a fix, I'll add more test cases

Reply:
in `python/letsql/backends/let/__init__.py`'s `Backend._register_and_transform_cache_tables` we use a conditional `if hasattr(node, "parent")` rather than `__recreate__(kwargs)`. Should we use the same thing in both places?

Reply:
Yes, we should I open a [PR](https://github.com/letsql/letsql/pull/44) refactoring  `_register_and_transform_cache_tables` including the use of `__recreate__(kwargs)` and other goodies